handlers.go

```diff
@@ -0,0 +1,293 @@
+						return
+					}
+				} else {
+					task.Date = time.Now().Format(DateFormat)
+				}
+			}
+		}
+		// Ловим пустую дату
+		if task.Date == "" {
+			task.Date = time.Now().Format(DateFormat)
+		}
+
+		ID := AddTask(task)
+		response := map[string]string{"id": fmt.Sprintf("%v", ID)}
+		w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+		err = json.NewEncoder(w).Encode(response)
+		if err != nil {
+			sendError(w, "Ошибка json")
+		}
+
+	case r.Method == "GET":
+		id := r.FormValue("id")
+		err := IDChecker(id)
+		if err != nil {
+			sendError(w, err.Error())
+			return
+		}
+		task, err := GetTaskByID(id)
+		if err != nil {
+			sendError(w, err.Error())
+			return
+		}
+
+		w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+		err = json.NewEncoder(w).Encode(task)
+		if err != nil {
+			sendError(w, "Ошибка json")
+		}
+
+	case r.Method == "PUT":
+		var task Task
+
+		err := json.NewDecoder(r.Body).Decode(&task)
+		if err != nil {
+			sendError(w, "json error")
+			return
+		}
+
+		err = IDChecker(task.ID)
+		if err != nil {
+			sendError(w, err.Error())
+			return
+		}
+
+		_, err = GetTaskByID(task.ID)
+		if err != nil {
+			sendError(w, err.Error())
+			return
+		}
+
+		// Ловим пустой заг
+		if task.Title == "" {
+			sendError(w, "Не указан заголовок задачи")
+			return
+		}
+		// Проверяем репит через NextDate
+		if task.Repeat != "" {
+			_, err := NextDate(Now, Now.Format(DateFormat), task.Repeat)
+			if err != nil {
+				sendError(w, err.Error())
+				return
+
+			}
+		}
+		// Ловим не корректную дату
+		if task.Date != "" {
+			taskDate, err := time.Parse(DateFormat, task.Date)
+			if err != nil {
+				sendError(w, "Неверная дата")
+				return
+			}
+			if taskDate.Before(time.Now()) {
+				if task.Repeat != "" {
+					task.Date, err = NextDate(time.Now(), task.Date, task.Repeat)
+					if err != nil {
+						sendError(w, err.Error())
+						return
+					}
+				} else {
+					task.Date = time.Now().Format(DateFormat)
+				}
+			}
+		}
+		// Ловим пустую дату
+		if task.Date == "" {
+			task.Date = time.Now().Format(DateFormat)
+		}
+		err = UpdateTaskByID(task)
+		if err != nil {
+			sendError(w, err.Error())
+			return
+		}
+
+		w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+		w.Write([]byte("{}"))
+
+	case r.Method == "DELETE":
+		id := r.FormValue("id")
+		err := IDChecker(id)
+		if err != nil {
+			sendError(w, err.Error())
+			return
+		}
+
+		_, err = GetTaskByID(id)
+		if err != nil {
+			sendError(w, err.Error())
+			return
+		}
+		err = DeleteTaskByID(id)
+		if err != nil {
+			sendError(w, err.Error())
+			return
+		}
+
+		w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+		w.Write([]byte("{}"))
+
+	default:
+		http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
+
+	}
+}
+
+// TasksHandler Handler нескольких задач. Методы: GET
+func TasksHandler(w http.ResponseWriter, r *http.Request) {
+	switch {
+	case r.Method == "GET":
```

Comment:
Если один случай, то можно заменить на 
```suggestion
	if r.Method != http.MethodGet {
		http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
	}
```

Reply:
Это мвп, поэтому оставил залог на добавление методов

Reply:
Да в обычном коде на уровне роутера распределяют ручки и метод запроса, так что в функцию приходит только определенный запрос, а не все сразу, как в случае с `/api/tasks`